File: mobile/ios/Air/SubModules/WalletCore/Models/CardBackground.swift
import Foundation

public enum CardBackground: String, CaseIterable, Codable, Identifiable, Sendable {
    case `default`
    case orange
    case green
    case sea
    case purple
    case pink
    case red

    public var id: String { rawValue }

    public var label: String {
        switch self {
        case .default:
            "Yohi"
        case .orange:
            "Blue"
        case .green:
            "Green"
        case .sea:
            "Ocean"
        case .purple:
            "Purple"
        case .pink:
            "Pink"
        case .red:
            "Red"
        }
    }

    public var imageName: String {
        switch self {
        case .default:
            "CardBackgroundDefault"
        case .orange:
            "CardBackgroundBlue"
        case .green:
            "CardBackgroundGreen"
        case .sea:
            "CardBackgroundOcean"
        case .purple:
            "CardBackgroundPurple"
        case .pink:
            "CardBackgroundPink"
        case .red:
            "CardBackgroundRed"
        }
    }
}

File: mobile/android/air/SubModules/UIComponents/src/main/java/org/mytonwallet/app_air/uicomponents/commonViews/CardBackground.kt
package org.mytonwallet.app_air.uicomponents.commonViews

import android.graphics.Color
import org.mytonwallet.app_air.uicomponents.R

enum class CardBackground(
    val id: String,
    val label: String,
    val imageRes: Int,
    val primaryTextColor: Int = Color.WHITE,
    val secondaryTextColor: Int = Color.argb(191, 255, 255, 255),
) {
    DEFAULT("default", "Yohi Wallet", R.drawable.img_card_default),
    ORANGE("orange", "Blue", R.drawable.img_card_blue),
    GREEN("green", "Green", R.drawable.img_card_green),
    SEA("sea", "Ocean", R.drawable.img_card_sea),
    PURPLE("purple", "Purple", R.drawable.img_card_purple),
    PINK("pink", "Pink", R.drawable.img_card_pink),
    RED("red", "Red", R.drawable.img_card_red);

    companion object {
        fun fromId(id: String?): CardBackground {
            return entries.firstOrNull { it.id == id } ?: DEFAULT
        }
    }
}
